### ISSUE TITLE
Update ResultAggregator to implement AutoCloseable.

### ISSUE THREAD

**Issue by JedS6391** [JedS6391/LGP#38]
It would be nice to not have to call `ResultAggregator::close()` explicitly when done with an aggregator instance.

If the `AutoCloseable` interface is implemented then the aggregator can be used with the `use` function by passing a block as a lambda. Then the aggregator will be automatically closed -- which is a much nicer API.

If `use` is not used then one can still call the `close()` function manually if they wish.

**Comment by JedS6391** [comment]
As an example, here is the current usage vs. proposed usage:

```
results.forEach { result -> 
    aggregator.add(result)
}

// Done aggregating results
aggregator.close()
```

vs.

```
aggregator.use { a ->
    results.forEach { result -> 
        a.add(result)
    }
}

// Aggregator is automatically closed.
```

**Comment by JedS6391** [comment]
This issue was closed when PR #39 was merged into `develop`.
